test/Examples.hs

```diff
@@ -78,17 +103,51 @@ rubySkips = Path.relFile <$>
   , "ruby_spec/core/enumerable/shared/inject.rb"
   -- Doesn't parse
   , "ruby_spec/language/string_spec.rb"
+  , "ruby_spec/language/fixtures/freeze_magic_comment_required_diff_enc.rb"
 
   -- Can't detect method calls inside heredoc bodies with precise ASTs
   , "ruby_spec/core/argf/readpartial_spec.rb"
   , "ruby_spec/core/process/exec_spec.rb"
   ]
 
+tsxSkips :: [Path.RelFile]
+tsxSkips = Path.relFile <$>
+  [
+  -- Cannot decode byte '\xe2': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
```

Comment:
Almost all languages have this same kind of encoding problem with certain example files that contain non-utf8 characters. Should we just leave this as a known issue for now?

Reply:
cc @maxbrunsfeld

Reply:
What is the error? Tree-sitter *should* just treat this as a syntax error; it shouldn't halt the whole thing.

Reply:
`Cannot decode byte '\xe2': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream` is the error, it's coming from [unmarshalling](https://github.com/tree-sitter/haskell-tree-sitter/blob/master/tree-sitter/src/TreeSitter/Unmarshal.hs#L124) in haskell-tree-sitter.

Reply:
Oh, I see. @robrix @aymannadeem For that Text-unmarshalling step, can we do some kind of *lossy* UTF8 decoding, e.g. in which invalid byte sequences are replaced by the [replacement character](https://en.wikipedia.org/wiki/Specials_(Unicode_block)#Replacement_character). I don't think we want to bail out of analyzing the whole file if there are some non-utf8 bytes somewhere.

Reply:
Yes!

``` haskell
decodeUtf8With lenientDecode
```

Reply:
Oooh, my bad, I misread the error. Fortunately @tclem fixed this yesterday!